go/ssa/interp/value.go

```diff
@@ -489,9 +489,35 @@ func (it *stringIter) next() tuple {
 	return okv
 }
 
-type mapIter chan [2]value
+type mapIter struct {
+	iter *reflect.MapIter
+	ok   bool
+}
+
+func (it *mapIter) next() tuple {
+	it.ok = it.iter.Next()
+	if !it.ok {
+		return []value{false, nil, nil}
+	}
+	k, v := it.iter.Key().Interface(), it.iter.Value().Interface()
+	return []value{true, k, v}
+}
 
-func (it mapIter) next() tuple {
-	kv, ok := <-it
-	return tuple{ok, kv[0], kv[1]}
+type hashmapIter struct {
+	iter *reflect.MapIter
+	ok   bool
+	cur  *entry
+}
+
+func (it *hashmapIter) next() tuple {
+	if !it.ok {
+		return []value{false, nil, nil}
+	}
+	if it.cur == nil {
+		it.ok = it.iter.Next()
+		it.cur = it.iter.Value().Interface().(*entry)
```

Comment:
If it.iter is exhausted, then it.ok==false and it.cur==null here and the next line will fail. I think you need to check it.ok and return false before this line.

Reply:
reworked this in 5bba70b